CRUD_Operations.c

```diff
@@ -0,0 +1,173 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define FILE_NAME "users.txt"
+
+typedef struct {
+    int id;
+    char name[50];
+    int age;
+} User;
+
+//for adding new users
+void create_user(){
+     User user;
+    FILE *file = fopen(FILE_NAME, "a");
+    if (!file) {
+        printf("Error opening file.\n");
+        return;
+    }
+
+    printf("Enter User ID: ");
+    scanf("%d", &user.id);
+    printf("Enter Name: ");
+    scanf(" %[^\n]", user.name);  // we can use this for taking string with spaces
+    printf("Enter Age: ");
+    scanf("%d", &user.age);
+    // adding data to our file
+    fprintf(file, "%d,%s,%d\n", user.id, user.name, user.age);
+    fclose(file);
+
+    printf("Successfully Added\n");
+};
+
+// for just viewing the data
+void read_users(){
+     FILE *file = fopen(FILE_NAME, "r");
+    if (!file) {
+        printf("Error opening file.\n");
+        return;
+    }
+
+    User user;
+    printf("\n**** User Data ****\n");
+    while (fscanf(file, "%d,%49[^,],%d\n", &user.id, user.name, &user.age) != EOF) {
```

Comment:
Explain what is the use of "%49[^,]"

Reply:
Okay so this is a format specifier used for reading lines till we get this ','. Basically, 49 is maximum number of characters to read the string. Even if we can change this number, We are using it so it does not get overflow.
And [^,] it is like we can read any character but if we get ' , ' then it will consider it as the next data. As while we are reading the file the file data is separated by ' , ' only.